src/Commands/PassCommand.cpp

```diff
@@ -1,30 +1,32 @@
 #include "PassCommand.hpp"
+#include "Client.hpp"
+#include "Server.hpp"
+#include <cctype>
 
 PassCommand::PassCommand(std::string source, std::vector<std::string> params,
                          Client* client) {
-    if (!source.empty()) {
-        throw std::invalid_argument("PASS command should not have a prefix.");
-    }
-    if (params.size() != 1) {
-        throw std::invalid_argument(
-            "PASS command requires exactly one parameter.");
-    }
-    this->_password = params[0];
-
-    if (_password.empty()) {
-        throw std::invalid_argument("Password cannot be empty.");
+    if (params.size() < 1) {
+        client->sendMessage(Replies::ERR_NEEDMOREPARAMS(client, "PASS"));
+        throw;
     }
-    for (std::string::iterator it = _password.begin(); it != _password.end();
-         ++it) {
-        if (!std::isprint(*it) || std::isspace(*it)) {
-            throw std::invalid_argument(
-                "Password contains invalid characters.");
-        }
+    if (!source.empty()) {
+        throw;
     }
-
     this->_client = client;
+    this->_password = params[0];
 };
 
 PassCommand::~PassCommand(){};
 
-void PassCommand::run() { return; }
\ No newline at end of file
+void PassCommand::run() {
```

Comment:
you didn't account for the fact that the server password was not set
if the password is empty, you need to return without sending any message.